app/src/main/java/space/taran/arknavigator/mvp/model/repo/PreviewRepo.kt

```diff
@@ -0,0 +1,56 @@
+package space.taran.arknavigator.mvp.model.repo
+
+import space.taran.arknavigator.mvp.model.dao.ResourceId
+import space.taran.arknavigator.mvp.model.dao.computeId
+import space.taran.arknavigator.ui.fragments.utils.PredefinedIcon
+import space.taran.arknavigator.ui.fragments.utils.Preview
+import space.taran.arknavigator.utils.*
+import java.nio.file.Files
+import java.nio.file.Path
+
+class PreviewRepo {
+
+    fun providePreview(path: Path, resourceID: ResourceId? = null): Preview {
+        if (Files.isDirectory(path)) {
+            return Preview(predefinedIcon = PredefinedIcon.FOLDER)
+        }
+
+        return createPair(path, resourceID)
+    }
+
+    private fun createPair(file: Path, resourceID: ResourceId?): Preview {
```

Comment:
I guess, name of this method left from some older code. Anyway, "Pair" is too ambiguous noun in the name.